packages/formStructure/components/FieldListContext.tsx

```diff
@@ -46,12 +46,11 @@ export const FieldListProvider: React.FC<FieldListProviderProps> = ({
 
   const removeListItem = (rowIndex: number) => {
     if (onRemoveItem) {
-      return onRemoveItem(rowIndex);
+      const handleRemove = onRemoveItem(rowIndex);
```

Comment:
I think this is a breaking change.

The previous functionality returned a curried function, calling the function directly changes the behavior. 

Additionally not returning and calling the callback and `setData` changes the expected behavior as well since as previously written if you set `onRemoveItem` the parent component was responsible for updating the data. This is how the component is used in several places.

Reply:
Check it out again, though. I only changed the handling within `FieldListContext`. `FieldList` still accepts a curried function. Is `FieldListContext` exposed as a component or just a part of `FieldList`?

Reply:
After looking again I'm pretty sure that FieldListContext is only used inside of FieldList, so we're free to refactor as long as the props on FieldList don't change.

Reply:
FieldListAddButton uses it as well, but only the addListItem.

Reply:
🤷‍♂️  tested this and seems to work and resolve the issue.

Reply:
I am still trying to understand this so I might be off base but I don't think we are actually using removeItem as a curried function anywhere on kommander-ui so removing this would still work. If someone did want to access the SyntheticEvent within a function (which is why it is probably curried), this might limit them from doing so, so it is a breaking change.

You can test this by editing line: 
https://github.com/mesosphere/kommander-ui/blob/master/kommander/frontend/src/components/molecules/FormLabelRow.tsx#:~:text=removeItem%3A%20(affectedIndex%3A%20number)%20%3D%3E%20()%20%3D%3E%20void%3B

to be `removeItem: (affectedIndex: number) => (event: React.SyntheticEvent) => void;`

and lines: 
https://github.com/mesosphere/kommander-ui/blob/master/kommander/frontend/src/pages/projects/ProjectForm.tsx#:~:text=removeItem%3D%7B(rowIndex)%20%3D%3E%20()%20%3D%3E%20%7B,%7D%7D

to be 
` removeItem={(rowIndex) => (event) => {
              console.log("event", event);
              onChange({
                namespaceLabels: removeIndex(
                  formData.namespaceLabels,
                  rowIndex
                ),
              });
            }}`
and notice that event is now undefined with these changes. That being said, I am not sure we need to access the event and to me it might be a bit of overkill. Happy to mess with it more once these other tasks are done.

Reply:
If I could get an example of a render failure and steps to recreate I could understand better.

Reply:
@clintonmedbery here is an error: https://github.com/dcos-labs/ui-kit/pull/608